src/schematron/schut-to-xspec.xsl

```diff
@@ -185,7 +185,7 @@
     <xsl:mode name="make-predicate" on-multiple-match="fail" on-no-match="fail" />
 
     <xsl:template match="@location" as="text()" mode="make-predicate">
-        <xsl:text expand-text="yes">[(${x:known-UQName('x:context')}/root()/({.}) treat as node()) is {x:known-UQName('x:select-node')}(${x:known-UQName('x:context')}/root(), @location, preceding-sibling::{x:known-UQName('svrl:ns-prefix-in-attribute-values')}, {parent::element() => x:xslt-version()})]</xsl:text>
+        <xsl:text expand-text="yes">[if (not((${x:known-UQName('x:context')}/root()/({.})) instance of node() and ({x:known-UQName('x:select-node')}(${x:known-UQName('x:context')}/root(), @location, preceding-sibling::{x:known-UQName('svrl:ns-prefix-in-attribute-values')}, {parent::element() => x:xslt-version()})) instance of node())) then error(QName('http://www.w3.org/2005/xqt-errors', 'XPDY0050'), 'ERROR in location (XPath should point to only one node "{.}")') else (${x:known-UQName('x:context')}/root()/({.}) treat as node()) is {x:known-UQName('x:select-node')}(${x:known-UQName('x:context')}/root(), @location, preceding-sibling::{x:known-UQName('svrl:ns-prefix-in-attribute-values')}, {parent::element() => x:xslt-version()})]</xsl:text>
```

Comment:
Putting a user-provided string into `''` (the 2nd parameter of `error()`) without escaping its `'` is fragile. These tests are now broken:
```console
$ bin/xspec.sh -s test/issue-396.xspec
...
Running Tests...
Syntax error near {...t()[. eq 'above-mentioned']...} at char 703 in expression in xsl:sequence/@select on line 250 column 46 of issue-396-compiled.xsl:
  XPST0003: expected ")", found name "above-mentioned"
Syntax error near {...t()[. eq 'above-mentioned']...} at char 703 in expression in xsl:sequence/@select on line 256 column 43 of issue-396-compiled.xsl:
  XPST0003: expected ")", found name "above-mentioned"
Errors were reported during stylesheet compilation

*** Error running the test suite
$ bin/xspec.sh -s test/schematron-026.xspec 
...
Running Tests...
Syntax error near {...sec-title-text-lower-case']...} at char 676 in expression in xsl:sequence/@select on line 215 column 46 of schematron-026-compiled.xsl:
  XPST0003: expected ")", found name "sec-title-text-lower-case"
...
*** Error running the test suite
$ bin/xspec.sh -s test/xspec-sch.xspec
...
Running Tests...
Syntax error near {...ith(@label, 'In function-pa...} at char 903 in expression in xsl:sequence/@select on line 225 column 46 of xspec-sch-compiled.xsl:
  XPST0003: expected ")", found name "In"
...
*** Error running the test suite
```

I'm fixing it.

Reply:
Fixed in e7084cf47009e230429e51fba59be3b2cf5367b8.